### PR TITLE
Timeouts configuration for haproxy backends

### PR DESCRIPTION
**Pull request by jsoriano**
Some code added to handle timeouts configuration:
* Backend timeouts per-port can be declared now as an annotation of the service.
* Default timeouts parametrized using environment variables for haproxy in the example docker.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,13 +122,44 @@ kind: Service
 metadata:
   annotations:
     kube2lb/port-mode: |
-      { "mysql": "tcp" }
+      { "mysql": "tcp", "healthz": "http" }
```
Comment (jsoriano) [inline README.md:125]: Not related. Modified example to have multiple ports, as it has caused confusion in the past.

File: README.md [hunk 2]
```diff
@@ -122,13 +122,44 @@ kind: Service
...
+their names in order to use this feature.
+
+They can be used in templates as an attribute of each service:
+
+```
+{{- range $i, $service := $services }}
+{{- if gt $service.Timeout 0 }}
+timeout server {{ $service.Timeout }}
+{{- end }}
+[[- end }}
```
Comment (mrodm) [inline README.md]: Typo in the last end tag. Should be `{{- end }}`

**Review by kamunas — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (jsoriano) [reply README.md]: Ouch, fixed, thanks!

File: kubernetes.go [hunk 3]
```diff
@@ -147,6 +148,15 @@ func (c *KubernetesClient) ExecuteTemplates(info *ClusterInformation) {
 	}
 }
 
+func (c *KubernetesClient) readAnnotation(meta v1.ObjectMeta, annotation string, value interface{}) {
+	if data, ok := meta.Annotations[annotation]; ok && len(data) > 0 {
```
Comment (dpaneda) [inline kubernetes.go]: I know this is not from this review, but this line is too dense to correctly read it. I would split it like:

```
if data, ok := meta.Annotations[annotation]
if ok && len(data) > 0 {
```

Which reads much better IMO

**Review by dpaneda — APPROVED** [review]

File: kubernetes.go (hunk 3 above)
Comment (jsoriano) [reply kubernetes.go]: Agree, changed.

**Review by mrodm — APPROVED** [review]

**Pull request merged**